### DISCUSSION TITLE
Pitch Reference Tracking mode

### DISCUSSION THREAD

**Discussion by rhaghi** [NatLabRockies/ROSCO#352] [Ideas]
Hi,

I have started working with ROSCO to tune a controller for a wind turbine based on the SCADA measurement. It has been an interesting journey, and I have learned a lot. Thank you very much for such a useful tool. 

I activate PRC in DISCON.IN will dictate the RPM to the controller or the rotor per wind speed. I wonder if it is possible to have a similar setting for the pitch. I mean defining pitch set points for the controller, and the turbine should go through them. I guess that would provide the possibility of tuning the aerodynamics for the specific controller setting. 

Let me know what you think.

Best,
Rad

**Comment by dzalkind** [comment]
Hi Rad,

For now, your best option would be to prescribe the minimum pitch vs wind speed like the peak shaving module does: https://github.com/NREL/ROSCO/blob/7deb583d4acbacc068bad28e2228f7d257a7cd7c/Examples/Test_Cases/IEA-15-240-RWT-UMaineSemi/DISCON-UMaineSemi.IN#L135

You could also define the pitch in open loop: https://github.com/NREL/ROSCO/blob/7deb583d4acbacc068bad28e2228f7d257a7cd7c/Examples/Test_Cases/IEA-15-240-RWT-UMaineSemi/DISCON-UMaineSemi.IN#L155

Right now, the open loop pitch versus time is the input, but it's not a huge jump to make the breakpoints wind speed.  This has been on our to-do list for a while, but there are no active plans to implement it.  Also, because it's not really how a wind turbine controller should work, but it has applications like you suggest.

If you're able to implement a change like this, we'd be happy to help integrate it.

Best, Dan
